src/Behat/Testwork/ServiceContainer/ExtensionManager.php

```diff
@@ -34,30 +32,50 @@
     private $debugInformation = array(
         'extensions_list' => array()
     );
+    /**
+     * @var ExtensionInstantiator[]
+     */
+    private $instantiators = array();
 
     /**
      * Initializes manager.
      *
-     * @param Extension[] $extensions     List of default extensions
-     * @param null|string $extensionsPath Base path where to search custom extension files
+     * @param Extension[]             $extensions      List of default extensions
+     * @param string|null             $extensionsPath  Base path where to search custom extension files (deprecated)
+     * @param ExtensionInstantiator[] $instantiators   Extension instantiators to use
      */
-    public function __construct(array $extensions, $extensionsPath = null)
+    public function __construct(array $extensions, $extensionsPath = null, array $instantiators = array())
     {
         foreach ($extensions as $extension) {
             $this->extensions[$extension->getConfigKey()] = $extension;
         }
 
-        $this->extensionsPath = $extensionsPath;
+        // BC Layer for `extensionsPath`
+        // to be removed in 4.0
+        if (empty($instantiators)) {
+            $instantiators = array(
+                new ExtensionClassInstantiator(),
+                new ExtensionFileInstantiator($extensionsPath),
+            );
+        }
```

Comment:
could be written as :

```php
$this->instantiators = empty($instantiators)
    ? array(new ExtensionClassInstantiator(), new ExtensionFileInstantiator($extensionsPath))
    : $instantiators;
```

But IMO this is not elegant (that's pretty subjective), even though it would comply with scrutinizer...